core/metrics/src/main/scala/org/typelevel/otel4s/metrics/Observable.scala

```diff
@@ -0,0 +1,7 @@
+package org.typelevel.otel4s.metrics
+
+import org.typelevel.otel4s.Attribute
+
+trait Observable[F[_], A] {
```

Comment:
The `Observable` name could be too broad. Also, it may clash with `java.util.Observable`. 

What if we use `ObservableMeasurement` there?